bookshelf/api_v3/ec2.py

```diff
@@ -0,0 +1,321 @@
+    security_groups = field(type=PVector, mandatory=True,
+                            factory=_parse_unicode_pvector)
+
+
+def _connect_to_ec2(region, credentials):
+    """
+    :param region: The region of AWS to connect to.
+    :param EC2Credentials credentials: The credentials to use to authenticate
+        with EC2.
+
+    :return: a connection object to AWS EC2
+    """
+    conn = boto.ec2.connect_to_region(
+        region,
+        aws_access_key_id=credentials.access_key_id,
+        aws_secret_access_key=credentials.secret_access_key
+    )
+    if conn:
+        return conn
+    else:
+        log_red('Failure to authenticate to EC2.')
+        return False
+
+
+def _create_server_ec2(connection,
+                       region,
+                       disk_name,
+                       disk_size,
+                       ami,
+                       key_pair,
+                       instance_type,
+                       tags={},
+                       security_groups=None,
+                       delete_on_termination=True,
+                       log=False,
+                       wait_for_ssh_available=True):
+    """
+    Creates EC2 Instance
+    """
+
+    if log:
+        log_green("Started...")
+        log_yellow("...Creating EC2 instance...")
+
+    ebs_volume = EBSBlockDeviceType()
+    ebs_volume.size = disk_size
+    bdm = BlockDeviceMapping()
+    bdm[disk_name] = ebs_volume
+
+    # get an ec2 ami image object with our choosen ami
+    image = connection.get_all_images(ami)[0]
+    # start a new instance
+    reservation = image.run(1, 1,
+                            key_name=key_pair,
+                            security_groups=security_groups,
+                            block_device_map=bdm,
+                            instance_type=instance_type)
+
+    # and get our instance_id
+    instance = reservation.instances[0]
+
+    #  and loop and wait until ssh is available
+    while instance.state == u'pending':
+        if log:
+            log_yellow("Instance state: %s" % instance.state)
+        sleep(10)
+        instance.update()
+    if log:
+        log_green("Instance state: %s" % instance.state)
+    if wait_for_ssh_available:
+        wait_for_ssh(instance.public_dns_name)
+
+    # update the EBS volumes to be deleted on instance termination
+    if delete_on_termination:
+        for dev, bd in instance.block_device_mapping.items():
+            instance.modify_attribute('BlockDeviceMapping',
+                                      ["%s=%d" % (dev, 1)])
+
+    # add a tag to our instance
+    connection.create_tags([instance.id], tags)
+
+    if log:
+        log_green("Public dns: %s" % instance.public_dns_name)
+
+    # returns our new instance
+    return instance
+
+
+@implementer(ICloudInstance)
+@provider(ICloudInstanceFactory)
+class EC2():
+    """
+    Class representing an EC2 instance, that provides methods for interacting
+    with the instance.
+
```

Comment:
maybe name it EC2instance ?
There are many types of objects in EC2 that are not just EC2 instances (ssh groups, key pairs, ELBs...) while bookshelf doesn't provide a method for managing those today (as it hasn't been needed so far). That is likely to happen in the future.
If/When that happens a class named EC2 would be imply too generic